.github/workflows/ruby-rspec.yml

```diff
@@ -0,0 +1,23 @@
+name: hcloud-ruby ci
+on: [push]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: [ '2.7' ]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+      - name: Run double tests
+        run: bundle exec rspec -t doubles --order rand
+      - name: Run legacy tests
+        run: LEGACY_TESTS=y bundle exec rspec -t ~doubles
+      - name: Run rubocop
```

Comment:
I would move this to a second job. I don't fancy mixing lint and spec. Either use the predefined Rubocop-Action or just do something like:
```yaml
jobs:
  lint:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        ruby-version: [ '2.7' ]
   steps:
      - uses: actions/checkout@v2
      - name: Setup Ruby
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ matrix.ruby-version }}
          bundler-cache: true
      - name: Run rubocop 
        run: bundle exec rubocop
```

You could even use a slimmer base-image than ubuntu and probably also skip the bundler part. Just install and execute rubocop withouth anything else (like we do it in some other projects)

Reply:
I've also added 3.0.